### PR TITLE
Allow setting Map config in environment

### PR DESCRIPTION
**Pull request by jimmed**
Hope you don't mind me raising a PR -- this docker image is absolutely ideal for my purposes, except for this one missing config item.

### CODE REVIEW

**Comment by Renegade-Master** [comment]
Hey, thanks for the contribution!

Unfortunately due to Ubuntu updating, you ae actually blocked on this until I merge #31 as you can see from the Checks.

**Comment by Renegade-Master** [comment]
If possible, could you add a test for this field to the validation GitHub Action? 

If you don't understand it, that's fine, but it will have to be added by one of us 🙂

**Comment by jimmed** [comment]
Thanks for the speedy reply! I'm not sure exactly what kind of validation you would like me to add, but I'm happy to hold off until the blocking PR is merged anyway.

**Comment by Renegade-Master** [comment]
Like this:
1. Set the custom properties to something other than their default - [Link](https://github.com/Renegade-Master/zomboid-dedicated-server/blob/main/.github/workflows/docker-build.yml#L89)
2. Add a Check to the end like the existing ones to read the key, and expect a value - [Link](https://github.com/Renegade-Master/zomboid-dedicated-server/blob/main/.github/workflows/docker-build.yml#L209)

I will try to get that PR merged in the meantime though.
Thanks again!

**Comment by Renegade-Master** [comment]
I rebased this PR after I (finally) fixed the underlying issues in the images.
Currently passing the checks now 🎉, but I would still like some things to be added to this if possible:

- [ ] Add a test as described [above](https://github.com/Renegade-Master/zomboid-dedicated-server/pull/33#issuecomment-1208474961)
- [ ] Add a "sane default" in the event that the Environment Variable was not added/specified like the [other value initialisation](https://github.com/Renegade-Master/zomboid-dedicated-server/blob/main/src/run_server.sh#L179)
- [ ] Add documentation about the new Environment Variable in the [README.md](https://github.com/Renegade-Master/zomboid-dedicated-server/blob/main/README.md?plain=1#L141)

**Comment by jimmed** [comment]
Thanks for this!

I'll add in the missing default values and documentation and push those up shortly.
However, I would argue that you don't want `Map` to appear in the config in your CI tests:
- valid values for this config are tied to the contents of workshop mods
	- missing maps could block server startup
	- invalid mod IDs will block startup
	- broken map mods will block startup
- Mod-related config isn't already included in the tests

If you still want me to add those tests, I'm happy to, but I don't know how we could work around the dependency on workshop content.

**Comment by jimmed** [comment]
...and fixed the mistake I made in the very first commit :sweat_smile: ADHD moment.

File: src/run_server.sh
Comment (Renegade-Master) [inline src/run_server.sh:194]: Considering that `Muldraugh, Ky` is the default according to the [Documentation](https://pzwiki.net/wiki/Server_Settings), maybe we should set it to that here.

```suggestion
    MAP_NAMES=${MAP_NAMES:-"Muldraugh, Ky"}
```

**Review by Renegade-Master — CHANGES_REQUESTED** [review]
One small thing. Might as well use the default configuration if none is given, rather than simply overwriting it to blank.

File: README.md
Comment (Renegade-Master) [inline README.md:155]:
```suggestion
| `MAP_NAMES`           | Map Names (e.g. North;South)                                             | map1;map2;map3    | Muldraugh, KY |
```

**Comment by Renegade-Master** [comment]
Looks like the maps were requested, [but not found](https://github.com/Renegade-Master/zomboid-dedicated-server/runs/7752575244?check_suite_focus=true#step:7:1388).
image

Did you add those files yourself?

**Comment by Renegade-Master** [comment]
@jimmed 

Ahh, I see.
image

[Full logs](https://github.com/Renegade-Master/zomboid-dedicated-server/suites/7738435603/artifacts/324628268)

So, you can only select Maps that you have added, right? We aren't adding any Maps so it can't find them.

Should be easy enough to download the maps during the Action though, right?

**Comment by jimmed** [comment]
> @jimmed 
> 
> Ahh, I see.
> image
> 
> [Full logs](https://github.com/Renegade-Master/zomboid-dedicated-server/suites/7738435603/artifacts/324628268)
> 
> So, you can only select Maps that you have added, right? We aren't adding any Maps so it can't find them.
> 
> Should be easy enough to download the maps during the Action though, right?

Indeed, we'd need to have a mod from the workshop added which we could guarantee exists and contains a map with those names.

It might be easier to omit this from the test entirely, _or_ provide an unlisted map mod in the workshop that we can reference

**Review by theCrius — APPROVED** [review]

**Comment by Renegade-Master** [comment]
@jimmed 
Can you link me to a Map that you intend to use? Preferably the smallest map you know to minimise download times.

I'll see if I can stick together a test.

**Comment by jimmed** [comment]
> @jimmed Can you link me to a Map that you intend to use? Preferably the smallest map you know to minimise download times.
> 
> I'll see if I can stick together a test.

The smallest map we're currently using is probably [this one](https://steamcommunity.com/sharedfiles/filedetails/?id=2721653030). Thanks for all of your hard work on this!

**Comment by Renegade-Master** [comment]
Ah, damnit. I duplicated one of your commits.
Luckily this will be squashed when rebasing the PR, so it won't really matter.

**Comment by Renegade-Master** [comment]
@jimmed 
Can you test if this works in your usecase? Alternatively, I can just merge this, release a new version, and you can find out that way.

**Review by Renegade-Master — APPROVED** [review]
Looks good to me 👍

Even though it is partially my work anyway 😅

**Comment by jimmed** [comment]
Hey! Thanks for this! I'm not in a position to test this out until next week, but I'll spin up an instance with our config and see if it works as expected. It _should_ work fine, based on the above tests you've added :rocket:

**Comment by Renegade-Master** [comment]
Ah, we'll just give it a go so.

**Pull request merged**
